### PR TITLE
Add ability score builder UI

### PR DESCRIPTION
**Pull request by fryguy1013**
Builds on previous PR to have a UI. Based on work from Mats

image

image

image

### CODE REVIEW

**Comment by TMunsie** [comment]
Closes #2211

File: src/module/actor/character/index.ts [hunk 1]
```diff
@@ -939,6 +969,13 @@ class CharacterPF2e extends CreaturePF2e {
                 : { rank: 0, value: 0 };
         })();
 
+        // Expose the higher between highest spellcasting DC and (if present) class DC
+        this.data.data.attributes.classOrSpellDC = ((): { rank: number; value: number } => {
+            const spellDC = this.data.data.attributes.spellDC ?? { rank: 0, value: 0 };
+            const { classDC } = this.data.data.attributes;
+            return spellDC.value > classDC.value ? { ...spellDC } : { rank: classDC.rank, value: classDC.value };
+        })();
+
```
Comment (CarlosFdez) [inline src/module/actor/character/index.ts:978]: Seems this slipped in again somehow. This feature already exists on characters.

Comment (stwlam) [inline src/module/actor/character/index.ts:978]:
```suggestion
```

File: src/module/actor/character/index.ts [hunk 2]
```diff
@@ -274,7 +274,7 @@ class CharacterPF2e extends CreaturePF2e {
     /** Setup base ephemeral data to be modified by active effects and derived-data preparation */
     override prepareBaseData(): void {
         super.prepareBaseData();
-        const systemData: DeepPartial<CharacterSystemData> = this.data.data;
+        const systemData: DeepPartial<CharacterSystemData> & { abilities: object } = this.data.data;
```
Comment (stwlam) [inline src/module/actor/character/index.ts:277]:
```suggestion
        const systemData: DeepPartial<CharacterSystemData> & { abilities: Abilities } = this.data.data;
```

File: src/styles/actor/character/_ability-builder.scss [hunk 3]
```diff
@@ -0,0 +1,404 @@
+.loot-actor-popup .window-content {
```
Comment (stwlam) [inline src/styles/actor/character/_ability-builder.scss:1]: ??

File: static/lang/en.json [hunk 4]
```diff
@@ -64,6 +64,27 @@
...
+            "AbilityScoreMethod": {
+                "Title": "Ability Score Method",
+                "Description": "Switching method resets your ability score distribution",
+                "UseCustomLabel": "Use Custom"
+            },
+            "AncestryMissingHelp": "Choose am ancestry to select boosts",
+            "BackgroundMissingHelp": "Choose a background to select boosts",
+            "ClassMissingHelp": "Choose a class to select boosts",
+            "NotSelected": "Not Selected"
+        },
```
Comment (stwlam) [inline static/lang/en.json:87]: Move to Actor.Character

File: static/template.json [hunk 5]
```diff
@@ -884,6 +866,10 @@
             "flaws": {
                 "0": { "value": [] }
             },
+            "voluntaryBoosts": {
+            },
+            "voluntaryFlaws": {
+            },
```
Comment (stwlam) [inline static/template.json:872]:
```suggestion
            "voluntaryBoosts": {},
            "voluntaryFlaws": {},
```

**Comment by stwlam** [comment]
Testing with Fumbus, it only recognizes my four free boosts, and the single key ability score option remains unselected and unselectable:

image

**Comment by fryguy1013** [comment]
fixed by reverting your change

**Pull request merged**